ext/eventcart/CRM/Event/Cart/BAO/MerParticipant.php

```diff
@@ -53,10 +57,18 @@ public static function create(&$params) {
       'status_id' => self::get_pending_in_cart_status_id(),
       'contact_id' => $params['contact_id'],
       'event_id' => $params['event_id'],
-      'cart_id' => $params['cart_id'],
     ];
     $participant = CRM_Event_BAO_Participant::create($participantParams);
-    $mer_participant = new CRM_Event_Cart_BAO_MerParticipant($participant);
+    if (!empty($params['cart_id']) && !EventCartParticipant::get(FALSE)
```

Comment:
@eileenmcnaughton why not use the save action instead?

Reply:
@seamuslee001 cos I would need to know the ID to do a do an update I think - which = doing a get first

Reply:
I would have thought you could do this right?

```
EventCartParticipant::save(FALSE)
   ->setValues(['cart_id' => $params['cart_id'], 'participant_id' => $participant->id])
   ->setMatch(['cart_id', 'participant_id'])
  ->execute();
```

Reply:
oh yeah that might work

Reply:
(I didn't know about `setMatch()`